docs/Formula-Cookbook.md

```diff
@@ -696,6 +696,16 @@ Homebrew provides two formula DSL methods for launchd plist files:
 * [`plist_name`](https://www.rubydoc.info/github/Homebrew/brew/master/Formula#plist_name-instance_method) will return e.g. `homebrew.mxcl.<formula>`
 * [`plist_path`](https://www.rubydoc.info/github/Homebrew/brew/master/Formula#plist_path-instance_method) will return e.g. `/usr/local/Cellar/foo/0.1/homebrew.mxcl.foo.plist`
 
+### Using environment variables
+
+Homebrew has multiple levels of environment variable filtering which impact which variables are available to formulae.
+
+Firstly, the overall environment in which Homebrew runs is [filtered](https://github.com/Homebrew/brew/blob/master/bin/brew#L54-L96) to avoid environment contamination breaking from-source builds ([ref](https://github.com/Homebrew/brew/issues/932)).  In particular, this process filters all but the given whitelisted variables relating to the shell, terminal, OS, proxies and so on, but allows anything prefixed with `HOMEBREW_` or related to CI and Travis through.
+
+The second level of filtering [removes "sensitive" environment variables](https://github.com/Homebrew/brew/blob/4897ec02d0f00c37a21ddaa0bebff6b83b27d027/Library/Homebrew/extend/ENV.rb#L32) to avoid malicious subprocesses obtaining them ([ref](https://github.com/Homebrew/brew/pull/2524)).  This has the effect of preventing any such variables from reaching a formula's Ruby code as they are filtered before it is reached.  This filtering remove variables that match the given case-insensitive pattern, which is to say that environment variables containing the names `COOKIE`, `KEY`, `TOKEN` or `PASSWORD` will be removed.
```

Comment:
Again, would you mind removing:
```
`COOKIE`, `KEY`, `TOKEN` or `PASSWORD`
```
and pointing to the code instead?

Similarly, you've linked to a specific commit here which means the line number will remain correct but unfortunately it'll not be up to date. Might want to reference a function name or similar instead?

Reply:
Yep, for sure.  I'll link to the file itself as a whole and reference the function name.